### PR TITLE
implement env variable expansion for kaniko builds

### PR DESCRIPTION
**Pull request by DanielSel**
Adds environment variable expansion for kaniko builds (to address currently inconsistent behaviour between buildArgs and envs).

### CODE REVIEW

File: pkg/skaffold/build/cluster/kaniko.go [hunk 1]
```diff
@@ -102,3 +115,74 @@ func (b *Builder) copyKanikoBuildContext(ctx context.Context, workspace string,
...
+		return nil, err
+	}
+	var generatedEnvs []v1.EnvVar
+	generatedEnvs = append(generatedEnvs, v1.EnvVar{Name: "IMAGE_REPO", Value: repoStr})
+	generatedEnvs = append(generatedEnvs, v1.EnvVar{Name: "IMAGE_NAME", Value: nameStr})
+	generatedEnvs = append(generatedEnvs, v1.EnvVar{Name: "IMAGE_TAG", Value: tagStr})
+	return generatedEnvs, nil
+}
+
+func parseImageParts(imageStr string) (repo, name, tag string, err error) {
```
Comment (tejal29) [inline pkg/skaffold/build/cluster/kaniko.go:170]: I am not sure i understand why we need this? We are adding extra env variables inferred from image built which are used in probably in your `Dockerfile`?
Can you elaborate more on your use case?

Thanks
Tejal

Comment (Awem) [reply pkg/skaffold/build/cluster/kaniko.go:170]: See https://github.com/GoogleContainerTools/skaffold/issues/3229 for background information. There are quite a few use cases for having information about the image tag inside your Dockerfile. For example: you might need some kind of version tag for the application inside your docker container

**Comment by DanielSel** [comment]
@tejal29 I think the CI re-build is stuck

**Comment by gsquared94** [comment]
@DanielSel It seems that Travis is stuck. You can push an empty commit to force it to trigger another job. That worked for me once before.

**Comment by tejal29** [comment]
@DanielSel you need to run `make generate-schema` on this branch.
Can you please do that?

**Comment by DanielSel** [comment]
Hey guys,
sorry I was unable to access my computer for quite a while now. I will do it as soon as I can find the time. Thank you!

**Comment by gsquared94** [comment]
@DanielSel np, I'll close the PR for now. Please reopen once you've made the change.

**Comment by DanielSel** [comment]
I made the change and re-submitted the PR here: #4557

File: pkg/skaffold/build/cluster/kaniko_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+func TestEnvInterpolation(t *testing.T) {
+	imageStr := "why.com/is/this/such/a/long/repo/name/testimage:testtag"
+	artifact := &latest.KanikoArtifact{
+		Env: []v1.EnvVar{{Name: "hui", Value: "buh"}},
+	}
+	generatedEnvs, err := generateEnvFromImage(imageStr)
+	if err != nil {
+		t.Fatalf("error processing generated env variables from image uri: %s", err)
+	}
+	env, err := evaluateEnv(artifact.Env, generatedEnvs...)
```
Comment (joanfabregat) [inline pkg/skaffold/build/cluster/kaniko_test.go:37]: The documentation in `v2beta3.json` states that the env vars are available for `buildArgs:` but it seems that is it only the case for `env:`. Should I report a bug?

**Pull request closed without merging**